cloudstack 4.13v unable to start management server

@rhtyd
@wido  <!--
Verify first that your issue/request is not already reported on GitHub.
Also test if the latest release and master branch are affected too.
Always add information AFTER of these HTML comments, but no need to delete the comments.
-->

##### ISSUE TYPE
<!-- Pick one below and delete the rest -->

 * Other

##### COMPONENT NAME
<!--
cloudstack management 
-->
~~~
cloudstack management 
~~~

##### CLOUDSTACK VERSION
<!--
New line separated list of affected versions, commit ID for issues on master branch.
-->

~~~
4.13
~~~

##### CONFIGURATION
<!--
Information about the configuration if relevant, e.g. basic network, advanced networking, etc.  N/A otherwise
-->
cloudstack 4.13v , hypervisor : KVM

##### OS / ENVIRONMENT
<!--

-->
centos7

##### SUMMARY
<!-- Explain the problem/feature briefly -->
I'm trying to setup cloudstack 4.13v, till two days back everything was working fine, but from last night,, when I formatted and re-tried to install cloudstack management  using the cmd : **cloudstack-setup-management** I'm getting the error : **ImportError: No module named dns.reslover** . And unable to start cloudstack management server.

##### STEPS TO REPRODUCE
<!--
For bugs, show exactly how to reproduce the problem, using a minimal test-case. Use Screenshots if accurate.
For new features, show how the feature would be used.
-->

<!-- Paste example playbooks or commands between quotes below -->
~~~
cloudstack-setup-management
[Screen Shot 2020-01-17 at    @@@12.47..zip](https://github.com/apache/cloudstack/files/4075765/Screen.Shot.2020-01-17.at.12.47.zip)

~~~

<!-- You can also paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!-- What did you expect to happen when running the steps above? -->

~~~
to install cloudstack management without facing above issue
~~~

##### ACTUAL RESULTS
<!-- What actually happened? -->

<!-- Paste verbatim command output between quotes below -->
~~~
cloudstack-setup-management
Traceback (most recent call last):
  File "/usr/bin/cloudstack-setup-management", line 24, in <module>
    from cloudutils.serviceConfigServer import cloudManagementConfig
  File "/usr/lib64/python2.7/site-packages/cloudutils/serviceConfigServer.py", line 17, in <module>
    from db import Database
  File "/usr/lib64/python2.7/site-packages/cloudutils/db.py", line 20, in <module>
    import mysql.connector
  File "/usr/lib64/python2.7/site-packages/mysql/connector/__init__.py", line 41, in <module>
    import dns.resolver
ImportError: No module named dns.resolver
~~~


grep -i -E 'exception|unable|fail|invalid|leak|warn|error' /var/log/cloudstack/management/management-server.log
2020-01-16 16:09:31,558 DEBUG [c.c.u.d.ScriptRunner] (main:null) (logid:) CREATE TABLE `cloud`.`s3` ( `id` bigint unsigned NOT NULL AUTO_INCREMENT COMMENT 'id', `uuid` varchar(40), `access_key` varchar(20) NOT NULL COMMENT ' The S3 access key', `secret_key` varchar(40) NOT NULL COMMENT ' The S3 secret key', `end_point` varchar(1024) COMMENT ' The S3 host', `bucket` varchar(63) NOT NULL COMMENT ' The S3 host', `https` tinyint unsigned DEFAULT NULL COMMENT ' Flag indicating whether or not to connect over HTTPS', `connection_timeout` integer COMMENT ' The amount of time to wait (in milliseconds) when initially establishing a connection before giving up and timing out.', `max_error_retry` integer COMMENT ' The maximum number of retry attempts for failed retryable requests (ex: 5xx error responses from services).', `socket_timeout` integer COMMENT ' The amount of time to wait (in milliseconds) for data to be transfered over an established, open connection before the connection times out and is closed.', `created` datetime COMMENT 'date the s3 first signed on', PRIMARY KEY (`id`), CONSTRAINT `uc_s3__uuid` UNIQUE (`uuid`) ) ENGINE=InnoDB DEFAULT CHARSET=utf8 
2020-01-16 16:09:37,681 DEBUG [c.c.u.d.ScriptRunner] (main:null) (logid:) CREATE VIEW `cloud`.`volume_view` AS select volumes.id, volumes.uuid, volumes.name, volumes.device_id, volumes.volume_type, volumes.size, volumes.created, volumes.state, volumes.attached, volumes.removed, volumes.pod_id, account.id account_id, account.uuid account_uuid, account.account_name account_name, account.type account_type, domain.id domain_id, domain.uuid domain_uuid, domain.name domain_name, domain.path domain_path, projects.id project_id, projects.uuid project_uuid, projects.name project_name, data_center.id data_center_id, data_center.uuid data_center_uuid, data_center.name data_center_name, vm_instance.id vm_id, vm_instance.uuid vm_uuid, vm_instance.name vm_name, vm_instance.state vm_state, vm_instance.vm_type, user_vm.display_name vm_display_name, volume_host_ref.size volume_host_size, volume_host_ref.created volume_host_created, volume_host_ref.format, volume_host_ref.download_pct, volume_host_ref.download_state, volume_host_ref.error_str, disk_offering.id disk_offering_id, disk_offering.uuid disk_offering_uuid, disk_offering.name disk_offering_name, disk_offering.display_text disk_offering_display_text, disk_offering.use_local_storage, disk_offering.system_use, storage_pool.id pool_id, storage_pool.uuid pool_uuid, storage_pool.name pool_name, cluster.hypervisor_type, vm_template.id template_id, vm_template.uuid template_uuid, vm_template.extractable, vm_template.type template_type, resource_tags.id tag_id, resource_tags.uuid tag_uuid, resource_tags.key tag_key, resource_tags.value tag_value, resource_tags.domain_id tag_domain_id, resource_tags.account_id tag_account_id, resource_tags.resource_id tag_resource_id, resource_tags.resource_uuid tag_resource_uuid, resource_tags.resource_type tag_resource_type, resource_tags.customer tag_customer, async_job.id job_id, async_job.uuid job_uuid, async_job.job_status job_status, async_job.account_id job_account_id from `cloud`.`volumes` inner join `cloud`.`account` ON volumes.account_id = account.id inner join `cloud`.`domain` ON volumes.domain_id = domain.id left join `cloud`.`projects` ON projects.project_account_id = account.id left join `cloud`.`data_center` ON volumes.data_center_id = data_center.id left join `cloud`.`vm_instance` ON volumes.instance_id = vm_instance.id left join `cloud`.`user_vm` ON user_vm.id = vm_instance.id left join `cloud`.`volume_host_ref` ON volumes.id = volume_host_ref.volume_id and volumes.data_center_id = volume_host_ref.zone_id left join `cloud`.`disk_offering` ON volumes.disk_offering_id = disk_offering.id left join `cloud`.`storage_pool` ON volumes.pool_id = storage_pool.id left join `cloud`.`cluster` ON storage_pool.cluster_id = cluster.id left join `cloud`.`vm_template` ON volumes.template_id = vm_template.id left join `cloud`.`resource_tags` ON resource_tags.resource_id = volumes.id and resource_tags.resource_type = 'Volume' left join `cloud`.`async_job` ON async_job.instance_id = volumes.id and async_job.instance_type = 'Volume' and async_job.job_status = 0 
2020-01-16 16:09:38,952 ERROR [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Unable to upgrade the database
com.cloud.utils.exception.CloudRuntimeException: Problem with getting the ec attribute 
Caused by: java.lang.IllegalArgumentException: Can not set long field com.cloud.upgrade.dao.VersionVO.id to java.math.BigInteger
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
2020-01-16 16:09:38,955 WARN [o.a.c.s.m.c.ResourceApplicationContext] (main:null) (logid:) Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Failed to start bean 'cloudStackLifeCycle'; nested exception is com.cloud.utils.exception.CloudRuntimeException: Unable to upgrade the database
2020-01-16 16:09:38,955 WARN [o.e.j.w.WebAppContext] (main:null) (logid:) Failed startup of context o.e.j.w.WebAppContext@6e38921c{/client,file:///usr/share/cloudstack-management/webapp/,UNAVAILABLE}{/usr/share/cloudstack-management/webapp}
org.springframework.context.ApplicationContextException: Failed to start bean 'cloudStackLifeCycle'; nested exception is com.cloud.utils.exception.CloudRuntimeException: Unable to upgrade the database
Caused by: com.cloud.utils.exception.CloudRuntimeException: Unable to upgrade the database
Caused by: com.cloud.utils.exception.CloudRuntimeException: Problem with getting the ec attribute 
Caused by: java.lang.IllegalArgumentException: Can not set long field com.cloud.upgrade.dao.VersionVO.id to java.math.BigInteger
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)

--- a/debian/control
+++ b/debian/control
@@ -15,7 +15,7 @@ Description: A common package which contains files which are shared by several C
 
 Package: cloudstack-management
 Architecture: all
-Depends: ${python:Depends}, openjdk-8-jre-headless | java8-runtime-headless | java8-runtime | openjdk-9-jre-headless, cloudstack-common (= ${source:Version}), sudo, python-mysql.connector, libmysql-java, augeas-tools, mysql-client, adduser, bzip2, ipmitool, file, lsb-release, init-system-helpers (>= 1.14~)
+Depends: ${python:Depends}, openjdk-8-jre-headless | java8-runtime-headless | java8-runtime | openjdk-9-jre-headless, cloudstack-common (= ${source:Version}), sudo, python-mysql.connector, libmysql-java, augeas-tools, mysql-client, adduser, bzip2, ipmitool, file, gawk, iproute2, lsb-release, init-system-helpers (>= 1.14~), qemu-utils, python-dnspython
 Conflicts: cloud-server, cloud-client, cloud-client-ui
 Description: CloudStack server library
  The CloudStack management server

--- a/packaging/centos7/cloud.spec
+++ b/packaging/centos7/cloud.spec
@@ -74,6 +74,8 @@ Requires: mysql-connector-python
 Requires: ipmitool
 Requires: %{name}-common = %{_ver}
 Requires: iptables-services
+Requires: qemu-img
+Requires: python-dns
 Group:     System Environment/Libraries
 %description management
 The CloudStack management server is the central point of coordination,

